velero/schedule/keycloak-br-script-cm.yaml

```diff
@@ -0,0 +1,120 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: keycloak-br-configmap
+  labels:
+    foundationservices.cloudpak.ibm.com: keycloak-data
+data:
+  br_keycloak.sh: |
+    #!/usr/bin/env bash
+
+    # Licensed Materials - Property of IBM
+    # Copyright IBM Corporation 2023. All Rights Reserved
+    # US Government Users Restricted Rights -
+    # Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+    #
+    # This is an internal component, bundled with an official IBM product.
+    # Please refer to that particular license for additional information.
+
+    # ---------- Command arguments ----------
+
+    set -o errtrace
+    set -o errexit
+
+    MODE=$1
+    KEYCLOAK_NAMESPACE=$2
+    OPERATOR_NAMESPACE=$3
+    if [[ -z $OPERATOR_NAMESPACE ]]; then
+        OPERATOR_NAMESPACE=$KEYCLOAK_NAMESPACE
+    fi
+
+    BACKUP_DIR=/keycloak/keycloak-backup
+
+    function main {
+        if [[ $MODE == "backup" ]]; then
+            info "Mode set to backup, beginning backup process."
+            backup
+            success "Backup completed successfully."
+        else
+            info "Mode is set to restore, beginning restore process."
+            restore
+            success "Restore completed successfully."
+        fi
+    }
+
+    function backup {
+        mkdir -p $BACKUP_DIR/database
+        CNPG_PRIMARY_POD=`oc get cluster.postgresql.k8s.enterprisedb.io keycloak-edb-cluster -o jsonpath="{.status.currentPrimary}" -n $KEYCLOAK_NAMESPACE` && \
+        oc -n $KEYCLOAK_NAMESPACE exec -t $CNPG_PRIMARY_POD -c postgres -- mkdir -p /run/keycloak_backup && \
+        oc -n $KEYCLOAK_NAMESPACE exec -t $CNPG_PRIMARY_POD -c postgres -- pg_dump -v --username=postgres --dbname=keycloak -f /run/keycloak_backup/keycloak_db_backup.dump
```

Comment:
Does this mean we need to increase the PVC side, or use ephemeral storage mount for backup?